### ISSUE TITLE
Reproducing the ResNet50 benchmark

### ISSUE THREAD

**Issue by byronyi** [apple/tensorflow_macos#13]
Running [TensorFlow official model garden ResNet50-v1.5](https://github.com/tensorflow/models/tree/9d9456d221edd5f1e886d06b2599f6460e8de3e4/official/vision/image_classification/resnet#pretrained-models) gives the following error on an M1 Mac mini:

```
$ PYTHONPATH=$(PWD) python3 official/vision/image_classification/resnet/resnet_ctl_imagenet_main.py --use_synthetic_data
WARNING:tensorflow:Some requested devices in `tf.distribute.Strategy` are not visible to TensorFlow: /job:localhost/replica:0/task:0/device:GPU:0
W1119 18:36:12.865904 4373019968 cross_device_ops.py:1316] Some requested devices in `tf.distribute.Strategy` are not visible to TensorFlow: /job:localhost/replica:0/task:0/device:GPU:0
INFO:tensorflow:Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
I1119 18:36:12.866842 4373019968 mirrored_strategy.py:350] Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
I1119 18:36:12.867031 4373019968 resnet_ctl_imagenet_main.py:138] Training 1 epochs, each epoch has 40036 steps, total steps: 40036; Eval 1563 steps
I1119 18:36:13.539694 4373019968 controller.py:365] restoring or initializing model...
restoring or initializing model...
I1119 18:36:13.539777 4373019968 controller.py:371] initialized model.
initialized model.
I1119 18:36:13.540033 4373019968 controller.py:214] train | step:      0 | training until step 40036...
train | step:      0 | training until step 40036...
2020-11-19 18:36:13.597005: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
2020-11-19 18:36:13.597137: W tensorflow/core/platform/profile_utils/cpu_utils.cc:126] Failed to get CPU frequency: 0 Hz
WARNING:tensorflow:From /Users/byronyi/Develop/models/official/staging/training/grad_utils.py:73: Hints.__new__ (from tensorflow.python.distribute.collective_util) is deprecated and will be removed in a future version.
Instructions for updating:
use distribute.experimental.CommunicationOptions instead
W1119 18:36:14.379299 6121844736 deprecation.py:333] From /Users/byronyi/Develop/models/official/staging/training/grad_utils.py:73: Hints.__new__ (from tensorflow.python.distribute.collective_util) is deprecated and will be removed in a future version.
Instructions for updating:
use distribute.experimental.CommunicationOptions instead
2020-11-19 18:36:20.475765: I tensorflow/compiler/tf2mlcompute/kernels/mlc_subgraph_op.cc:545] Compute: Failed in processing TF graph while/body/_1/gradient_tape/while/truediv/MLCSubgraphOp_0_58 with error: Internal: PerformGradientPassNodeRoutine: Failed to find forward-pass output for node: while/body/_1/while/conv1/kernel/Regularizer/mul (error will be reported 5 times unless TF_MLC_LOGGING=1).
2020-11-19 18:36:20.819 python3[11839:1137514] *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArrayM objectAtIndexedSubscript:]: index 3 beyond bounds [0 .. 2]'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000018af45320 __exceptionPreprocess + 240
	1   libobjc.A.dylib                     0x000000018ac73c04 objc_exception_throw + 60
	2   CoreFoundation                      0x000000018b00c064 -[__NSCFString characterAtIndex:].cold.1 + 0
	3   CoreFoundation                      0x000000018aeb3a1c -[__NSCFString hasSuffix:] + 0
	4   MLCompute                           0x0000000193fab250 -[MLCTrainingGraph resultGradientTensorToUseByExecuteGradientForLayer:sourceIndex:incrementIntermediateIndex:] + 540
	5   MLCompute                           0x0000000193fac598 -[MLCTrainingGraph allocateGradientTensorsForLayersInGraph:] + 660
	6   MLCompute                           0x0000000193fad004 -[MLCTrainingGraph compileAndAllocateGradientTensorsForGraph:] + 140
	7   MLCompute                           0x0000000193fb5708 -[MLCTrainingGraph executeGradientWithBatchSize:options:outputsData:completionHandler:] + 960
	8   _pywrap_tensorflow_internal.so      0x0000000131df6f90 _ZN10tensorflow9mlcompute3ops13MLCSubgraphOp23ExecuteMLCTrainingGraphEPNS_15OpKernelContextEPNS1_27MLCSubgraphExecutionContextEj + 436
	9   _pywrap_tensorflow_internal.so      0x0000000131df6360 _ZN10tensorflow9mlcompute3ops13MLCSubgraphOp20ProcessMLCSubgraphOpEPNS_15OpKernelContextEPNS1_27MLCSubgraphExecutionContextE + 1028
	10  _pywrap_tensorflow_internal.so      0x0000000131df8b4c _ZN10tensorflow9mlcompute3ops13MLCSubgraphOp7ComputeEPNS_15OpKernelContextE + 1084
	11  libtensorflow_framework.2.dylib     0x0000000105b19384 _ZN10tensorflow12_GLOBAL__N_113ExecutorStateINS_15PropagatorStateEE7ProcessENS2_10TaggedNodeEx + 2868
	12  libtensorflow_framework.2.dylib     0x0000000105b1a650 _ZNSt3__110__function6__funcIZN10tensorflow12_GLOBAL__N_113ExecutorStateINS2_15PropagatorStateEE7RunTaskINS_6__bindIMS6_FvNS5_10TaggedNodeExEJPS6_RKS9_RxEEEEEvOT_EUlvE_NS_9allocatorISJ_EEFvvEEclEv + 80
	13  libtensorflow_framework.2.dylib     0x0000000105b924b4 _ZN5Eigen15ThreadPoolTemplIN10tensorflow6thread16EigenEnvironmentEE10WorkerLoopEi + 552
	14  libtensorflow_framework.2.dylib     0x0000000105b9218c _ZZN10tensorflow6thread16EigenEnvironment12CreateThreadENSt3__18functionIFvvEEEENKUlvE_clEv + 80
	15  libtensorflow_framework.2.dylib     0x0000000105b83a74 _ZN10tensorflow12_GLOBAL__N_17PThread8ThreadFnEPv + 104
	16  libsystem_pthread.dylib             0x000000018adcd06c _pthread_start + 320
	17  libsystem_pthread.dylib             0x000000018adc7da0 thread_start + 8
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Fatal Python error: Aborted

Thread 0x0000000104a6fd40 (most recent call first):
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/tensorflow/python/eager/execute.py", line 59 in quick_execute
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/tensorflow/python/eager/function.py", line 555 in call
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/tensorflow/python/eager/function.py", line 1918 in _call_flat
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/tensorflow/python/eager/function.py", line 2942 in __call__
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py", line 888 in _call
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py", line 828 in __call__
  File "/Users/byronyi/Develop/models/orbit/standard_runner.py", line 141 in train
  File "/Users/byronyi/Develop/models/orbit/controller.py", line 413 in _train_n_steps
  File "/Users/byronyi/Develop/models/orbit/controller.py", line 218 in train
  File "/Users/byronyi/Develop/models/orbit/controller.py", line 306 in train_and_evaluate
  File "official/vision/image_classification/resnet/resnet_ctl_imagenet_main.py", line 176 in run
  File "official/vision/image_classification/resnet/resnet_ctl_imagenet_main.py", line 190 in main
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/absl/app.py", line 251 in _run_main
  File "/Users/byronyi/.virtualenv/tf/lib/python3.8/site-packages/absl/app.py", line 303 in run
  File "official/vision/image_classification/resnet/resnet_ctl_imagenet_main.py", line 197 in <module>
Abort trap: 6
```

**Comment by JacopoMangiavacchi** [comment]
I quickly tested training ResNet with 10 blocks on M1 with some real cifar data.

It trained quite fast on M1 but I got a segmentation fault if I specify validation too.  See code below an commented line at the end.

Epoch 1/30
194/195 [============================>.] - ETA: 0s - loss: 2.5521 - acc: 0.1561[1]    31192 segmentation fault  python resnet.py

It train perfectly for several epoch if I do not pass the validation to the fit method.

-o-

```python
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
import numpy as np
import datetime as dt

(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()

train_dataset = tf.data.Dataset.from_tensor_slices((x_train, y_train)).batch(64).shuffle(10000)
train_dataset = train_dataset.map(lambda x, y: (tf.cast(x, tf.float32) / 255.0, y))
train_dataset = train_dataset.map(lambda x, y: (tf.image.central_crop(x, 0.75), y))
train_dataset = train_dataset.map(lambda x, y: (tf.image.random_flip_left_right(x), y))
train_dataset = train_dataset.repeat()

valid_dataset = tf.data.Dataset.from_tensor_slices((x_test, y_test)).batch(5000).shuffle(10000)
valid_dataset = valid_dataset.map(lambda x, y: (tf.cast(x, tf.float32) / 255.0, y))
valid_dataset = valid_dataset.map(lambda x, y: (tf.image.central_crop(x, 0.75), y))
valid_dataset = valid_dataset.repeat()

def res_net_block(input_data, filters, conv_size):
  x = layers.Conv2D(filters, conv_size, activation='relu', padding='same')(input_data)
  x = layers.BatchNormalization()(x)
  x = layers.Conv2D(filters, conv_size, activation=None, padding='same')(x)
  x = layers.BatchNormalization()(x)
  x = layers.Add()([x, input_data])
  x = layers.Activation('relu')(x)
  return x

def non_res_block(input_data, filters, conv_size):
  x = layers.Conv2D(filters, conv_size, activation='relu', padding='same')(input_data)
  x = layers.BatchNormalization()(x)
  x = layers.Conv2D(filters, conv_size, activation='relu', padding='same')(x)
  x = layers.BatchNormalization()(x)
  return x

inputs = keras.Input(shape=(24, 24, 3))
x = layers.Conv2D(32, 3, activation='relu')(inputs)
x = layers.Conv2D(64, 3, activation='relu')(x)
x = layers.MaxPooling2D(3)(x)

num_res_net_blocks = 10
for i in range(num_res_net_blocks):
  x = res_net_block(x, 64, 3)

x = layers.Conv2D(64, 3, activation='relu')(x)
x = layers.GlobalAveragePooling2D()(x)
x = layers.Dense(256, activation='relu')(x)
x = layers.Dropout(0.5)(x)
outputs = layers.Dense(10, activation='softmax')(x)

res_net_model = keras.Model(inputs, outputs)

res_net_model.compile(optimizer=keras.optimizers.Adam(),
              loss='sparse_categorical_crossentropy',
              metrics=['acc'])


res_net_model.fit(train_dataset, epochs=30, steps_per_epoch=195)

# res_net_model.fit(train_dataset, epochs=30, steps_per_epoch=195, validation_data=valid_dataset, validation_steps=1)
```

**Comment by anna-tikhonova** [comment]
@byronyi Thank you very much for reporting this issue. We will investigate and report back.

**Comment by byronyi** [comment]
Changing the model to [the one in Keras applications](https://keras.io/api/applications/resnet/) works. Not sure if it is the benchmark in [the blog post](https://blog.tensorflow.org/2020/11/accelerating-tensorflow-performance-on-mac.html).

**Comment by powderluv** [comment]
anyone else able to recreate the benchmarks on the blog post ? I get similar errors with tf_cnn_benchmark.py

**Comment by IuliaLungu-synthara** [comment]
Any developments on this? I am getting the same error.
